### PR TITLE
Stopping Error and Music still playing after deletion FIXED, and some extras.

### PR DESCRIPTION
**Pull request by MathijsvVelde**
Calling AV.Player.stop() after AV.Player.preload() but before AV.Player.play() results in this error:
```Uncaught TypeError: Cannot read property 'destroy' of undefined``` - FIXED
* [/src/device.coffee:38:16](https://github.com/MathijsvVelde/aurora.js/blob/dcb4f73d891726db05c486c2eb31dda2be4ada0f/src/device.coffee#L38)

Music still playing after deleting the AV.Player, AV.Player.asset or AV.Asset object.
Added the ability to destroy the AV.Player, AV.Player.asset or AV.Asset objects
* [/src/asset.coffee:153:5/5:97](https://github.com/MathijsvVelde/aurora.js/blob/dcb4f73d891726db05c486c2eb31dda2be4ada0f/src/asset.coffee#L153-L157)
* [/src/player.coffee:162:5/5:91](https://github.com/MathijsvVelde/aurora.js/blob/dcb4f73d891726db05c486c2eb31dda2be4ada0f/src/player.coffee#L162-L166)

Added the ability to remove all event handlers from specific event or an entire class. (needed for the above)
* [/src/core/events.coffee:10:9/9:276](https://github.com/MathijsvVelde/aurora.js/blob/dcb4f73d891726db05c486c2eb31dda2be4ada0f/src/core/events.coffee#L10-L18)

Updated the Speaker, just in case...

### CODE REVIEW

**Comment by MathijsvVelde** [comment]
Something wrong with the Travis check?

**Comment by crackofdusk** [comment]
Yeah, it's not correctly configured yet. I just switched on "Build only if .travis.yml is present" as a temporary solution until we fix that.

File: src/core/events.coffee [hunk 1]
```diff
@@ -7,9 +7,16 @@ class EventEmitter extends Base
...
+        return unless @events?
+        if @events?[event]
+            if fn?
+                index = @events[event].indexOf(fn)
+                @events[event].splice(index, 1) if ~index
+            else
+                @events[event]
+        else unless event?
+            events = {}
+        else return
```
Comment (fabienbrooke) [inline src/core/events.coffee]: do you need the last else

File: package.json [hunk 2]
```diff
@@ -1,44 +1,44 @@
...
+	"browserify" : {
+		"transform" : [
+			"coffeeify"
+		]
+	},
+	"browser" : {
+		"./node" : "./browser.coffee",
+		"./src/sources/node/file.coffee" : "./src/sources/browser/file.coffee",
+		"./src/sources/node/http.coffee" : "./src/sources/browser/http.coffee"
+	}
```
Comment (fabienbrooke) [inline package.json]: looks like you accidentally switched the spaces to tabs?

File: src/core/events.coffee (hunk 1 above)
Comment (MathijsvVelde) [inline src/core/events.coffee]: no i don't

**Comment by MathijsvVelde** [comment]
No, I didn't need that last `else`... …
And I set the indentation of 'package.json' back to 2 spaces indentation instead of tabs.
I find tabs easier for navigation with your arrow keys through your code.
But I don't change the 'package.json' file that often... so there you go! 

OK, that looks better!

**Comment by crackofdusk** [comment]
@MathijsvVelde Could you rebase to have 4 commits matching the 4 things you describe in your very first comment?

**Comment by MathijsvVelde** [comment]
You want me to create commits for each file I changed?
I don't think that's smart, except for `/src/device.coffee`.
But in `/src/device.coffee` I only added a `?`, that's all... no big deal I thought.
The changes I made in `/src/core/events.coffee` are needed for the changes I made in `/src/asset.coffee` and `/src/player.coffee`.
After that I only perfected `/src/core/events.coffee` a bit.

So can you tell me what I should do exactly?

**Comment by MathijsvVelde** [comment]
before I changed `/src/core/events.coffee`, using the `EventEmitter`.`off` function without arguments wouldn't do anything...

For example:
Before: 
```coffeescript
@off 'eventname', function # Removes the event listener "function" from event "eventname"
@off 'eventname' # does nothing
@off() # does nothing
```
After: 
```coffeescript
@off 'eventname', function # Removes the event listener "function" from event "eventname"
@off 'eventname' # Removes ALL event listeners from event "eventname"
@off() # Removes ALL event listeners from the emitting class
```

A bit like jQuery's function [.off()](http://api.jquery.com/off/)

**Comment by crackofdusk** [comment]
I meant grouping related changes (especially since there was some back and forth with indentation and coding style). In my opinion, the commits should be:

Bump speaker dependency version
Add ability to remove all handlers from a specific event
Fix playback not stopping after cleanup

Does that make sense?

**Comment by MathijsvVelde** [comment]
Like that?

**Comment by crackofdusk** [comment]
Yeah, but can you also clean the history so that only the 4 relevant commits remain? That involves something like:

```
$ git rebase -i 1e67c48
```

(1e67c48 is the last commit from the official repository before your commits)

That opens the following in your editor:
```
pick 47195f6 Bump speaker dependency version
pick 7a7d606 Add ability to remove all handlers from a specific event or class
pick c2b498f added the cleanup ability for AV.Player and AV.Asset
pick e67f65f Fix playback not stopping after cleanup
pick 380dd37 Calling AV.Player.stop() after AV.Player.preload() but before AV.Player.play() results in this error: ```Uncaught TypeError: Cannot read property 'destroy' of undefined``` - FIXED * [/src/device.coffee:38:16](/src/device.coffee#L38)
pick f92e93b It was a little to aggressive.
pick 0a13cff No, I didn't need that last `else`... And I set the indentation of 'package.json' back to 2 spaces indentation instead of tabs.
pick dcb4f73 and that thing to
```

Delete all lines after commit `e67f65f` so that only the following remains:
```
pick 47195f6 Bump speaker dependency version
pick 7a7d606 Add ability to remove all handlers from a specific event or class
pick c2b498f added the cleanup ability for AV.Player and AV.Asset
pick e67f65f Fix playback not stopping after cleanup
```

then save and quit. This will remove the bottom commits from the history. Then you just `git push --force` and this will update the pull request.

Thanks, and sorry for the nitpicking.

**Comment by MathijsvVelde** [comment]
mmm... rebasing in terminal is easier than in SourceTree GUI...
Yep, i'm still learning GitHub...

**Pull request merged**